staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -315,11 +315,24 @@ func ordinaryPromiseFactoryFactory(qs *queueSet) promiseFactory {
 	return promise.NewWriteOnce
 }
 
-// Seats returns the number of seats this request requires.
-func (req *request) Seats() int {
+// MaxSeats returns the maximum number of seats this request requires, it is
+// the maxumum of the two - WorkEstimate.InitialSeats, WorkEstimate.FinalSeats.
+func (req *request) MaxSeats() int {
+	return req.workEstimate.MaxSeats()
+}
+
+func (req *request) InitialSeats() int {
 	return int(req.workEstimate.InitialSeats)
 }
 
+func (req *request) AdditionalWorkSeatMicrosecond() int64 {
+	return req.workEstimate.AdditionalLatency.Microseconds() * int64(req.workEstimate.FinalSeats)
+}
+
+func (req *request) totalWorkSeatMicrosecond(estimatedServiceSeconds float64) float64 {
+	return estimatedServiceSeconds*float64(req.workEstimate.InitialSeats) + float64(req.AdditionalWorkSeatMicrosecond())
```

Comment:
This sum has mixed units.  The left term is seat-seconds, the right term is seat-microseconds.

Why introduce microseconds when the code was previously consistently using seconds?

Reply:
yes, waiting to rebase on top of https://github.com/kubernetes/kubernetes/pull/105322.

Reply:
Please now rebase on top of https://github.com/kubernetes/kubernetes/pull/105412

All these functions should now simply return SeatSeconds